src/auth/src/credentials/external_account.rs

```diff
@@ -467,4 +514,242 @@ mod test {
             }
         }
     }
+
+    #[tokio::test]
+    async fn test_external_account_with_impersonation_success() {
+        let subject_token_server = Server::run();
+        let sts_server = Server::run();
+        let impersonation_server = Server::run();
```

Comment:
oh wow, I have not been paying attention to @alvarowolfx's work. This is cool.

Reply:
Oh yeah httptest is cool! Cleaner than axum